File: .tmp_py4013/batch_17_dump/3876_construct_uniform_parity_array_ii.js
// LeetCode 3876 - Construct Uniform Parity Array II
// https://leetcode.com/problems/construct-uniform-parity-array-ii/

var uniformArray = function(nums1) {
    let mn = Infinity;
    for (const x of nums1) {
        if (x % 2 === 1 && x < mn) mn = x;
    }
    for (const x of nums1) {
        if (x % 2 === 0 && mn !== Infinity && x < mn) return false;
    }
    return true;
};

File: .tmp_rs4013/rust_check21/3876_construct_uniform_parity_array_ii.rs
struct Solution;
// LeetCode 3876 - Construct Uniform Parity Array II
// https://leetcode.com/problems/construct-uniform-parity-array-ii/

impl Solution {
    pub fn uniform_array(nums1: Vec<i32>) -> bool {
        let mut mn = i32::MAX;
        for &x in &nums1 {
            if x % 2 == 1 && x < mn {
                mn = x;
            }
        }
        for &x in &nums1 {
            if x % 2 == 0 && mn != i32::MAX && x < mn {
                return false;
            }
        }
        true
    }
}
